hedera-node/src/main/java/com/hedera/services/queries/validation/QueryFeeCheck.java

```diff
@@ -117,4 +121,89 @@ ResponseCodeEnum adjustmentPlausibility(AccountAmount adjustment) {
 
 		return OK;
 	}
+
```

Comment:
From line 124-209, not sure if there is any special purpose for this or not. If this simply for validate the payment transaction, I remember in the AnswerFlow class, we should already have a interface to fulfill this functionality.

Reply:
@ljianghedera If we don't specify `explicitPayment` a default transaction is generated. Else it uses the payment transaction we provide.

Reply:
@ljianghedera `StakedAnswerFlow-> satisfyUsing->validatePayment`  calls  the same method `validateTransactionPreConsensus` , to validate transactions before consensus. 

These are extra validations needed only for query payment transactions. So, apart from the validations done in `validateTransactionFeeCoverage` additional validations are added in `queryFeeCheck.validateQueryPaymentTransaction`